components/eamxx/src/physics/share/tests/physics_saturation_run_and_cmp.cpp

```diff
@@ -0,0 +1,320 @@
+#include "catch2/catch.hpp"
+
+#include "physics/share/physics_functions.hpp"
+#include "physics/share/physics_saturation_impl.hpp"
+#include "physics_unit_tests_common.hpp"
+
+#include "share/scream_types.hpp"
+#include "share/scream_session.hpp"
+#include "share/util/scream_utils.hpp"
+
+#include "ekat/ekat_pack.hpp"
+#include "ekat/util/ekat_file_utils.hpp"
+#include "ekat/util/ekat_test_utils.hpp"
+#include "ekat/kokkos/ekat_kokkos_utils.hpp"
+
+#include <thread>
+#include <array>
+#include <algorithm>
+#include <random>
+#include <iomanip>      // std::setprecision
+
+namespace scream {
+namespace physics {
+namespace unit_test {
+
+template <typename D>
+struct UnitWrap::UnitTest<D>::TestSaturation
+{
+  KOKKOS_FUNCTION  static void saturation_tests(
+    const Scalar& temperature, const Scalar& pressure,
+    Scalar& sat_ice_fp, Scalar& sat_liq_fp, Scalar& mix_ice_fr, Scalar& mix_liq_fr,
+    Scalar& sat_ice_mkp, Scalar& sat_liq_mkp, Scalar& mix_ice_mkr, Scalar& mix_liq_mkr)
+  {
+    //Nomenclature:
+    //subscript "_fp"  stands for "Flatau Pressure"
+    //subscript "_fr"  stands for "Flatau mixing Ratios"
+    //subscript "_mkp" stands for "Murphy Koop Pressure"
+    //subscript "_mkr" stands for "Murphy Koop mixing Ratios"
+
+    //Allow usage of saturation functions
+    using physics = scream::physics::Functions<Scalar, Device>;
+
+    //Convert Scalar inputs to Spacks because that's what polysvp1 and qv_sat expect as inputs.
+    //--------------------------------------
+    const Spack temps(temperature);
+    const Spack pres(pressure);
+
+    //Get values from polysvp1 and qv_sat (qv_sat calls polysvp1 here) to test against "expected" values
+    //--------------------------------------
+    sat_ice_fp  = physics::polysvp1(temps, true, Smask(true))[0];
+    sat_liq_fp  = physics::polysvp1(temps, false, Smask(true))[0];
+    //last argument "0" of qv_sat function below forces qv_sat to call "polysvp1"
```

Comment:
I think this comment about last arg being 0 and the comment ~8 lines down about last arg of 1 aren't true any more because you're passing physics::MurphyKoop and physics_Polysvp1 now? Delete them? Or is physics::MurphyKoop==1?

Reply:
Yes, looks like it's an enum now instead of a bool. The comments aren't needed since the enum name makes it clear what's going on. I will remove them.